src/languages/singlestoredb/singlestoredb.formatter.ts

```diff
@@ -0,0 +1,267 @@
+import { expandPhrases } from 'src/expandPhrases';
+import Formatter from 'src/formatter/Formatter';
+import Tokenizer from 'src/lexer/Tokenizer';
+import { EOF_TOKEN, isToken, type Token, TokenType } from 'src/lexer/token';
+import { keywords } from './singlestoredb.keywords';
+import { functions } from './singlestoredb.functions';
+
+const reservedCommands = expandPhrases([
+  // queries
+  'SELECT [ALL | DISTINCT | DISTINCTROW]',
+  'FROM',
+  'WHERE',
+  'GROUP BY',
+  'HAVING',
+  'ORDER BY',
+  'LIMIT',
+  'OFFSET',
+  // Data manipulation
+  // - insert:
+  'INSERT [IGNORE] [INTO]',
+  'VALUES',
+  'REPLACE [INTO]',
+  // - update:
+  'UPDATE',
+  'SET',
+  // - delete:
+  'DELETE [FROM]',
+  // - truncate:
+  'TRUNCATE [TABLE]',
+  // Data definition
+  'CREATE [ROWSTORE] [REFERENCE | TEMPORARY | GLOBAL TEMPORARY] TABLE [IF NOT EXISTS]',
+  'CREATE [OR REPLACE] [TEMPORARY] PROCEDURE [IF NOT EXISTS]',
+  'CREATE [OR REPLACE] [EXTERNAL] FUNCTION',
+  'DROP [TEMPORARY] TABLE [IF EXISTS]',
```

Comment:
Should also move `CREATE VIEW` into this section like in other `formatter.ts` files.